api/src/routes/email-subscription.test.ts

```diff
@@ -0,0 +1,265 @@
+        `${HOME_LOCATION}${urlEncodedInfoMessage2}`
+      );
+      expect(response.status).toBe(302);
+    });
+
+    test('should 302 redirect with info message if bad ID', async () => {
+      const response = await superRequest('/resubscribe/54321edcba', {
+        method: 'GET'
+      });
+      expect(response.headers.location).toStrictEqual(
+        `${HOME_LOCATION}${urlEncodedInfoMessage3}`
+      );
+      expect(response.status).toBe(302);
+    });
+
+    test("should set 'sendQuincyEmail' to 'true' for user with matching ID and 302 redirect with success message", async () => {
+      await fastifyTestInstance.prisma.user.createMany({
+        data: testUserData2
+      });
+
+      const response = await superRequest(`/resubscribe/${unsubscribeId1}`, {
+        method: 'GET'
+      });
+
+      const users = await fastifyTestInstance.prisma.user.findMany({
+        where: {
+          OR: [
+            { unsubscribeId: unsubscribeId1 },
+            { unsubscribeId: unsubscribeId2 }
+          ]
+        }
+      });
+
+      expect(users).toHaveLength(3);
+      users.forEach(user => {
+        if (user.unsubscribeId === unsubscribeId1) {
+          expect(user.sendQuincyEmail).toBe(true);
+        } else {
+          expect(user.sendQuincyEmail).toBe(false);
+        }
+      });
+
+      expect(response.headers.location).toStrictEqual(
+        `${HOME_LOCATION}${urlEncodedSuccessMessage2}`
+      );
+
+      expect(response.status).toBe(302);
+      await fastifyTestInstance.prisma.user.deleteMany({
+        where: {
+          OR: [
+            { unsubscribeId: unsubscribeId1 },
+            { unsubscribeId: unsubscribeId2 }
+          ]
+        }
+      });
+    });
+
+    test("should set 'sendQuincyEmail' to 'true' for first user with matching ID and 302 redirect with success message", async () => {
+      await fastifyTestInstance.prisma.user.createMany({
+        data: testUserData2
+      });
+
+      const response = await superRequest(`/resubscribe/${unsubscribeId2}`, {
+        method: 'GET'
+      });
+
+      const users = await fastifyTestInstance.prisma.user.findMany({
+        where: {
+          OR: [
+            { unsubscribeId: unsubscribeId1 },
+            { unsubscribeId: unsubscribeId2 }
+          ]
+        }
+      });
+
+      expect(users).toHaveLength(3);
+      users.forEach(user => {
+        if (user.email === 'user2@freecodecamp.org') {
+          expect(user.sendQuincyEmail).toBe(true);
+        } else {
+          expect(user.sendQuincyEmail).toBe(false);
+        }
+      });
+
+      expect(response.headers.location).toStrictEqual(
+        `${HOME_LOCATION}${urlEncodedSuccessMessage2}`
+      );
+
+      expect(response.status).toBe(302);
+      await fastifyTestInstance.prisma.user.deleteMany({
+        where: {
+          OR: [
+            { unsubscribeId: unsubscribeId1 },
+            { unsubscribeId: unsubscribeId2 }
+          ]
+        }
+      });
```

Comment:
Could you move this to an `afterEach` hook? It's a better separation of concerns if the test just tests stuff.